app/models/update_user.rb

```diff
@@ -0,0 +1,33 @@
+class UpdateUser < Struct.new(:params, :current_user)
+  attr_reader :status
+
+  def save
+    fail UnauthorizedError if !allowed_to_edit?
+
+    update_user
+  end
+
+  def errors
+    user.errors.full_messages.to_sentence
+  end
+
+  def user
+    @user ||= User.find(params[:id])
+  end
+
+  private
+
+  def update_user
+    user.assign_attributes(user_params)
+    SamAccountReckoner.new(user).clear
```

Comment:
Ah I see. This is also where we could put a check against the SAM API if we wanted to do that